io/disk/bonnie.py

```diff
@@ -46,21 +46,39 @@ def setUp(self):
         Source:
          http://www.coker.com.au/bonnie++/experimental/bonnie++-1.03e.tgz
         """
+
         fstype = self.params.get('fs', default='ext4')
-        smm = SoftwareManager()
-        deps = ['gcc', 'make']
-        if distro.detect().name == 'Ubuntu':
-            deps.extend(['g++'])
-        else:
-            deps.extend(['gcc-c++'])
-        if fstype == 'btrfs':
-            if distro.detect().name == 'Ubuntu':
-                deps.extend(['btrfs-tools'])
-
-        for package in deps:
-            if not smm.check_installed(package) and not smm.install(package):
-                self.cancel("Fail to install/check %s, which is needed for"
-                            "Bonnie test to run" % package)
+
+        if process.system("which bonnie++", ignore_status=True):
+            smm = SoftwareManager()
+            if not smm.check_installed('bonnie++') \
+                    and not smm.check_installed('bonnie++'):
+                '''Install the package from upstream'''
+                deps = ['gcc', 'make']
+                if distro.detect().name == 'Ubuntu':
+                    deps.extend(['g++'])
```

Comment:
Can you extend the `btrfs` check of L62 here?

Reply:
not get that i think that is suitable if condition can you help me to rectify this